tracker/tracker/src/main/modules/adoptedStyleSheets.ts

```diff
@@ -0,0 +1,155 @@
+import type App from '../app/index.js'
+import {
+  TechnicalInfo,
+  AdoptedSSReplaceURLBased,
+  AdoptedSSInsertRuleURLBased,
+  AdoptedSSDeleteRule,
+  AdoptedSSAddOwner,
+  AdoptedSSRemoveOwner,
+} from '../app/messages.gen.js'
+import { isRootNode } from '../app/guards.js'
+
+type StyleSheetOwner = (Document | ShadowRoot) & { adoptedStyleSheets: CSSStyleSheet[] }
+
+function hasAdoptedSS(node: Node): node is StyleSheetOwner {
+  return (
+    isRootNode(node) &&
+    // @ts-ignore
+    typeof node.adoptedStyleSheets !== 'undefined'
```

Comment:
can be just `node.adoptedStyleSheets` since undefined will default to false

Reply:
it can